### PR TITLE
Experiment: Replace `SiteEntity` with `SiteNodeAggregate`

### PR DESCRIPTION
**Pull request by mhsdesign**
WIP

Related: https://github.com/neos/neos-development-collection/issues/4470

only works for one site right now at time - no real domain support. 

For migration use 

```
flow site:setProperties neosdemo --siteResourcesPackageKey Neos.Demo --name "Neos Demo"
```

### CODE REVIEW

File: Neos.Neos/Classes/Domain/Model/Domain.php [hunk 1]
```diff
@@ -53,13 +53,6 @@ class Domain implements CacheAwareInterface
      */
     protected $port;
 
-    /**
-     * @var Site
-     * @ORM\ManyToOne(inversedBy="domains")
-     * @Flow\Validate(type="NotEmpty")
-     */
-    protected $site;
```
Comment (mhsdesign) [inline Neos.Neos/Classes/Domain/Model/Domain.php:61]: how should the relation ship of 

domain entity -> site node aggregate

work? Currently in this wip the domain wouldnt know its site directly but one has to fetch all site node aggregates and look into them

File: Neos.Neos/Classes/Domain/Repository/SiteRepository.php [hunk 2]
```diff
@@ -56,6 +56,36 @@ class SiteRepository extends Repository
...
     protected $defaultSiteNodeName;
 
+    public function findOnline(): iterable
+    {
+        return $this->findAll();
+    }
+
+    public function findAll(): iterable
+    {
+        $cr = $this->contentRepositoryRegistry->get(ContentRepositoryId::fromString('default'));
```
Comment (mhsdesign) [inline Neos.Neos/Classes/Domain/Repository/SiteRepository.php:106]: how can we find all site node aggregates efficiently? Loop over all crs registered and fetch the node aggregates und the `Neos.Neos:Sites` node?

Comment (kitsunet) [reply Neos.Neos/Classes/Domain/Repository/SiteRepository.php:106]: that seems SUPER expensive for something that will happen on every request. (we could cache it of course).

If anything, I would see a custom projection for it?

**Comment by mhsdesign** [comment]
Closing this experiment as written [here](https://github.com/neos/neos-development-collection/issues/4470#issuecomment-1846962521)

> While discussing this topic in [Slack](https://neos-project.slack.com/archives/C3MCBK6S2/p1701111101991739) we realized that the Node can't be the central authority for sites because there is no central Content Repository.

That _could_ be worked around by caching or projections or other stuff but it might not be worth the hassle. At least definitely not for now.

Also  working with node aggregates is not easy at all, https://github.com/neos/neos-development-collection/issues/4830 as we just want to use aggregate scoped properties but still have to issue the set properties command on a fake (random) dsp

For the neos 9 release we will keep the site entity and domain entity as is.

**Pull request closed without merging**